ports/esp8266/network_wlan.c

```diff
@@ -170,6 +170,38 @@ static mp_obj_t esp_disconnect(mp_obj_t self_in) {
 }
 static MP_DEFINE_CONST_FUN_OBJ_1(esp_disconnect_obj, esp_disconnect);
 
+static mp_obj_t build_stations_tuple(void) {
+    mp_uint_t count = wifi_softap_get_station_num();
+    if (count == 0) {
+        return mp_const_empty_tuple;
+    }
+    mp_obj_tuple_t *stations = NULL;
+    struct station_info *info = wifi_softap_get_station_info();
+    if (info == NULL) {
+        mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("no station info available"));
+    }
+    nlr_buf_t nlr;
+    if (nlr_push(&nlr) == 0) {
+        stations = MP_OBJ_TO_PTR(mp_obj_new_tuple(count, NULL));
+        struct station_info *current = info;
+        size_t station_index = 0;
+        mp_obj_t entry[2] = { 0 };
+        char bssid_str[18] = { 0 };
+        while (current != NULL && station_index < count) {
+            mp_uint_t bssid_len = snprintf(bssid_str, 18, "%02X:%02X:%02X:%02X:%02X:%02X", current->bssid[0], current->bssid[1], current->bssid[2], current->bssid[3], current->bssid[4], current->bssid[5]);
+            entry[0] = mp_obj_new_str(bssid_str, bssid_len);
```

Comment:
Suggest that this entry just be a bytes object of the 6-byte MAC.

Reply:
This is addressed in the latest version of the PR, thanks for pointing this out!